includes/Modules/Sign_In_With_Google.php

```diff
@@ -107,4 +110,51 @@ protected function setup_assets() {
 	protected function setup_settings() {
 		return new Settings( $this->options );
 	}
-}
+
+	/**
+	 * Renders the sign in button.
+	 *
+	 * @since n.e.x.t
+	 */
+	private function render_signin_button() {
+		$settings = $this->get_settings()->get();
+		if ( ! $settings['clientID'] ) {
+			return;
+		}
+
+		$redirect_url = site_url( '/auth/google' );
+		if ( substr( $redirect_url, 0, 5 ) !== 'https' ) {
+			return;
+		}
+
+		// Render the Sign in with Google button and related inline styles.
+		?>
+<!-- <?php echo esc_html__( 'Sign in with Google button added by Site Kit', 'google-site-kit' ); ?> -->
+<?php /* phpcs:ignore WordPress.WP.EnqueuedResources.NonEnqueuedScript */ ?>
+<script src="https://accounts.google.com/gsi/client" async></script>
+<div id="g_id_onload"
+	data-client_id="<?php echo esc_attr( $settings['clientID'] ); ?>"
+	data-login_uri="<?php echo esc_url( $redirect_url ); ?>"
+	data-auto_prompt="false">
+</div>
+<div class="g_id_signin"
+	style="display: none;"
+	data-type="standard"
+	data-size="large"
+	data-theme="<?php echo esc_attr( $settings['theme'] ); ?>"
+	data-text="<?php echo esc_attr( $settings['text'] ); ?>"
+	data-shape="<?php echo esc_attr( $settings['shape'] ); ?>"
+	data-logo_alignment="left">
+</div>
+<script type="text/javascript">
+window.addEventListener("load", () => {
+	if ( document.getElementsByClassName('g_id_signin')?.[0] ) {
+		document.getElementById('login').insertBefore(document.getElementsByClassName('g_id_signin')?.[0], document.getElementById('loginform'));
+		document.getElementsByClassName('g_id_signin')[0].style.display = 'block';
+	}
+});
```

Comment:
Thanks, @benbowler. Almost everything looks good, but I think we can make this code better if we create and insert everything using js:

```html
<script>
    ( () => {
        const onload = document.createElement( 'div' );
        onload.id = 'g_id_onload';
        onload.dataset.client_id = '<?php echo esc_js( $settings['clientID'] ); ?>';
        onload.dataset.login_url = '<?php echo esc_js( $redirect_url ); ?>';
        onload.dataset.auto_prompt = 'false';
        
        document.getElementById( 'login' ).insertBefore( onload, document.getElementById('loginform') );
        
        ...
        
        const lib = document.createElement( 'script' );
        ...
    } )();
</script>
```

This will let us avoid adding html and then move it to the right position. Once we do this, we will be good to go with this PR.

Reply:
Good idea, I switched to the official JS API which is much more clear.

Reply:
Awesome! Yeah, it is much better 👍